packages/material-ui/src/styles/experimentalStyled.js

```diff
@@ -59,7 +59,8 @@ const variantsResolver = (props, styles, theme, name) => {
   return variantsStyles;
 };
 
-const shouldForwardProp = (prop) => prop !== 'styleProps' && prop !== 'theme' && prop !== 'sx';
+const shouldForwardProp = (prop) =>
+  prop !== 'styleProps' && prop !== 'theme' && prop !== 'isRtl' && prop !== 'sx';
```

Comment:
would using a dictionary be faster to perform the test?

Reply:
I don't think the number of props is big enough in order for perf to win over the readability of the code. We could extract them to object if you really this can be an issue.